openscm/adapters/__init__.py

```diff
@@ -32,6 +31,9 @@ class Adapter(metaclass=ABCMeta):
     _initialized: bool
     """``True`` if model has been initialized via :func:`_initialize_model`"""
 
+    _initialized_inputs: bool
```

Comment:
I don't think we need this

Reply:
how else do you tell if you've initiliased inputs when making a call to `intialise_run_parameters`?

Reply:
Why do we need to store that for every run? The `_initialized` is only necessary for the `OpenSCM` class to only initialize the model once it is needed for the first time. `OpenSCM` makes sure that the methods are called in the right order.

Reply:
> The `_initialized` is only necessary for the `OpenSCM` class to only initialize the model once it is needed for the first time

Similarly, `_initialized_inputs` ensures that OpenSCM only sets the inputs which are expensive (e.g. writing input files in MAGICC) once. If `initialize_run_parameters` is public, don't we need this sort of check so users can't accidentally call `initialize_run_parameters` without calling `initialize_model_input`?